searx/query.py

```diff
@@ -306,28 +310,30 @@ def _parse_query(self):
             qlist.append(query_part)
             last_index_location = (qlist, len(qlist) - 1)
 
-        self.autocomplete_location = last_index_location
+        return last_index_location
 
-    def get_autocomplete_full_query(self, text):
+    def get_autocomplete_full_query(self, text: str) -> str:
+        assert self.autocomplete_location is not None
         qlist, position = self.autocomplete_location
         qlist[position] = text
         return self.getFullQuery()
 
-    def changeQuery(self, query):
-        self.user_query_parts = query.strip().split()
-        self.query = self.getFullQuery()
-        self.autocomplete_location = (self.user_query_parts, len(self.user_query_parts) - 1)
-        self.autocomplete_list = []
-        return self
+    def changeQuery(self, query: str) -> "RawTextQuery":
```

Comment:
> Suggestion provided by Dalf on https://github.com/searxng/searxng/issues/3337

Its not clear, why this method is named `changedQuery`:

- what does it change? .. and why should the caller change (a "immutable") query
- seems to be a factory method?  .. what does it produce?

My point is, when someone (like me) reads these lines of code its unclear what the intention of the code is and for what it is good .. I have to look at the blocks of code where this code is used to get a vague idea of what the intent of this code was

Unfortunately, this PR does nothing to change this situation, even if it may remedy a current issue.

- https://github.com/searxng/searxng/issues/3337#issuecomment-1999553069

Reply:
Happy to rename the method to something like 'copyWithNewQuery', and should solve the ambiguity.

The goal here is incremental development. Making this method immutable has its benefits, and we can have a subsequent PR to handle answerers. Perhaps this method will be deleted if no longer used then, but it can be immutable for the time being.

Reply:
On reading the implementations and a fresh night of sleep, I saw the same pattern `changeQuery` was being used for. Its only current use it to create a new FullQuery with a fresh set of parameters. However, since `getFullQuery` only requires state from `query_parts` and `user_query_parts`, the entire modification isn't necessary for an updated `getFullQuery`. Check out the new commits - it shows the current usages + what the code is good for now :smile: should reduce risk with this clearer usage.